### ISSUE TITLE
zip_list fails with large files

### ISSUE THREAD

**Issue by schuemie** [r-lib/zip#65]
I have a zip file containing a large file (> 4GB).  The zip file was created using the `zipr` function (`compression_level = 2`, although I'm not sure that matters). When I call `zip_list` I get 

```
Error in zip::zip_list("S:/SkeletonComparativeEffectStudy/cmOutput/CmData_l1_t1_c2_v2.zip") : 
  Cannot open zip file `S:\SkeletonComparativeEffectStudy\cmOutput\CmData_l1_t1_c2_v2.zip`
```
I can decompress the zip file just fine using `unzip` function. Other zip files that don't contain large files do work with `zip_list`.

This is a new development. The function was working a month ago. I'm currently on zip version 2.1.1, but it also happens on 2.1.0. My sessionInfo:

```
R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.0.2 tools_4.0.2    zip_2.1.1 
```

**Comment by gaborcsardi** [comment]
Do you think you could try to create a reproducible example? Thanks!

**Comment by schuemie** [comment]
Sure, I'll post that tomorrow. (Generating a large file using R code is *slow*).

FYI: the problem was not yet present at git tag [v2.0.4](https://github.com/r-lib/zip/releases/tag/v2.0.4)

**Comment by schuemie** [comment]
Here's the code that reproduces the problem. I discovered the problem only occurs when the resulting zip file is large (probably > 2GB,) not when the contents of the zip is large.

```r
# Generate large file. This takes 20 minutes on my system:
generatePoorlyCompressibleData <- function() {
  data.frame(x = runif(1e7), y = runif(1e7), z = runif(1e7))
}
write.table(generatePoorlyCompressibleData(), "data.txt")
addMore <- function(i) {
	write.table(generatePoorlyCompressibleData(), "data.txt", append = TRUE, col.names = FALSE)
}
invisible(sapply(1:8, addMore))

# Zip:
zip::zipr("big.zip", files = "data.txt", compression_level = 2)

# This fails:
zip::zip_list("big.zip")
# Error in zip::zip_list("big.zip") : 
# Cannot open zip file `C:\Users\mschuemi\Documents\big.zip`
```

**Comment by chris-english** [comment]
Not having same issue in Ubuntu. Was happy to rtm and notice `zip_list`

> zip_list_df_tst <- zip_list('test.zip')
> head(zip_list_df_tst)
        filename compressed_size uncompressed_size           timestamp
1 000316da7.flac         3252768           3251773 2020-11-18 01:24:08
2 003bc2cb2.flac         3517177           3516102 2020-11-18 01:24:08
3 0061c037e.flac         3378131           3377101 2020-11-18 01:24:12
4 010eb14d3.flac         3648907           3647792 2020-11-18 01:24:14
5 011318064.flac         3301025           3300015 2020-11-18 01:24:18
6 01307e693.flac         3509853           3508783 2020-11-18 01:24:20
  permissions
1         600
2         600
3         600
4         600
5         600
6         600
> class(zip_list_df_tst)
[1] "data.frame"
> class(zip_list_df_tst$filename)
[1] "character"
> class(zip_list_df_tst$compressed_size)
[1] "numeric"
> sum(zip_list_df_tst$compressed_size)/2^30
[1] 6.293851
> sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.1 LTS

Matrix products: default
BLAS:   /usr/local/lib/R/lib/libRblas.so
LAPACK: /usr/local/lib/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] zip_2.1.1

loaded via a namespace (and not attached):
[1] MASS_7.3-51.6  compiler_4.0.0 tools_4.0.0    seewave_2.1.6  monitoR_1.0.7 
[6] tuneR_1.3.3    signal_0.7-6   tcltk_4.0.0   
>

**Comment by gaborcsardi** [comment]
Fixed by #79.
